lib/libc/common/Kconfig

```diff
@@ -15,11 +15,28 @@ config COMMON_LIBC_MALLOC
 config COMMON_LIBC_MALLOC_ARENA_SIZE
 	int "Size of the common C library malloc arena"
 	depends on COMMON_LIBC_MALLOC
-	default 0
+	default 0 if MINIMAL_LIBC
+	default 16384 if MMU
```

Comment:
Does 16384 work for all arches with an MMU?

Reply:
This is inherited from the picolibc configuration, with the goal of not changing anything when using that. However, I note that picolibc also has `default 1024 if USERSPACE && MPU_REQUIRES_POWER_OF_TWO_ALIGNMENT`, which I didn't manage to copy over.

A couple of options here:

 1. Merge as-is, expecting people using a malloc heap will end up setting the size manually most of the time anyways.
 2. Do a test with the 'picolibc by default' and this branch merged together and see what breaks -- that would catch configurations expecting the above condition to be used.
 3. Add the above condition so that picolibc users don't see any change.

I'm leaning towards option 3. as that introduces the least change for existing picolibc users.

Reply:
Yeah, I thought that's what I had done before, but forgot that I removed the missing default clause when I had improved the common code to deal with MPU POT alignment requirements so that the missing default wasn't necessary anymore. We can revisit what the right defaults are when we've got more experience using this in more applications.